### ISSUE TITLE
Creating Filestore via Front End is broken due to missing required setting (network_id)

### ISSUE THREAD

**Issue by saltysoup** [GoogleCloudPlatform/cluster-toolkit#1096]
When creating a new filestore via the front end interface, this errors out as the FS module that is generated as part of the blueprint is missing a required setting (see below).
```
[root@ofe front-end]# cat fs/fs_11/ghpc_create_log.stderr
2023/03/29 21:51:13 failed to apply deployment variables in modules when expanding the config: a required setting is missing from a module: Module ID: small-fs Setting: network_id
```

This is the blueprint that is automatically generated by the front end (broken):
```
blueprint_name: small-fs

vars:
  project_id: injae-sandbox-340804
  deployment_name: small-fs
  region: asia-southeast1
  zone: asia-southeast1-c
  labels:
    created_by: fggf-server

deployment_groups:
- group: primary
  modules:
  - source: modules/file-system/filestore
    kind: terraform
    id: small-fs
    settings:
      filestore_share_name: home
      network_name: sweet-poodle-network
      zone: asia-southeast1-c
      size_gb: 1024
      filestore_tier: BASIC_HDD
    outputs:
    - network_storage
```

By manually editing the modules to add a network module instead of the `network_name` param in settings, the filestore can now deploy via front end.

```
blueprint_name: small-fs

vars:
  project_id: injae-sandbox-340804
  deployment_name: small-fs
  region: asia-southeast1
  zone: asia-southeast1-c
  labels:
    created_by: fggf-server

deployment_groups:
- group: primary
  modules:
  - source: modules/file-system/filestore
    kind: terraform
    id: small-fs
    settings:
      filestore_share_name: home
      network_name: sweet-poodle-network
      zone: asia-southeast1-c
      size_gb: 1024
      filestore_tier: BASIC_HDD
    outputs:
    - network_storage
```

## Question: Has there been changes to the ghpc validator to check if network module is used and enforced (in filestore module and other existing modules)?

**Comment by cboneti** [comment]
Hi Injae, 

Thanks for finding this. 

We did change filestore usage in [this PR](https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/969) as this would facilitate the usage of shared VPCs with filestore.  Because it did not break blueprints where filestore would `use` a network, we did not anticipate the issue. We inadvertently broke the front end. 

I believe the right thing to do would be that the front end created a blueprint with both a "pre-existing-vpc" and then used that vpc in filestore. That would allow for longer term compatibility. We will work with NAG to address that shortly.

**Comment by saltysoup** [comment]
Hi Carlos, that makes sense.

FWIW, the front end requires the VPC to exist before creating a file store so the auto generated blueprint would always use existing-vpc module instead of vpc (creating new)



On 30 Mar 2023, at 12:25 pm, Carlos Boneti ***@***.***> wrote:

﻿

Hi Injae,

Thanks for finding this.

We did change filestore usage in this PR<https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/969> as this would facilitate the usage of shared VPCs with filestore. Because it did not break blueprints where filestore would use a network, we did not anticipate the issue. We inadvertently broke the front end.

I believe the right thing to do would be that the front end created a blueprint with both a "pre-existing-vpc" and then used that vpc in filestore. That would allow for longer term compatibility. We will work with NAG to address that shortly.
